### DISCUSSION TITLE
A transformer to add prefix for each ultilities before process.

### DISCUSSION THREAD

**Discussion by sondh0127** [unocss/unocss#3579] [Q&A]
![image](https://github.com/unocss/unocss/assets/62163604/584334ba-6b2c-4f0b-8b08-2ac6b2d52415)

Hello everyone. I have a scenario which I wanted to migrate a lib component that used unocss without a prefix to a prefix that contained 'ui-'

I know that the group prefix like `ui-(flex flex-col)` would work, but it still take a lot of work.

Can anyone guide me a way to write a transformer(Vite) for adding prefix at correct ultilities. thanks

**Comment by zyyv** [comment]
This hook can help you. https://unocss.dev/config/#preprocess

```ts
preprocess: m => 'ui-' + m,
```

**Reply by efroim102** [reply]
@zyyv I have a similar case where before shipping a package, I'd like the Vite build process to add a prefix to all the CSS classes used in the application, both in the html templates and in the Unocss generated styles.

Adding the preprocess hook does not seem to work.

The package I'm developing is a Vue3 UI component library that depends on headless UI. I tried using the `vue-scoped` mode to scope the CSS to the component, unfortunately headless uses teleport for quite a few of its components (e.g. Dialog), resulting in the styles not being applied correctly as the generated scoped CSS classes no longer match the html hierarchy.
